pkg/setting/setting_dashboard_previews.go

```diff
@@ -0,0 +1,31 @@
+package setting
+
+import (
+	"time"
+
+	"gopkg.in/ini.v1"
+)
+
+type DashboardPreviewsSettings struct {
+	SchedulerInterval time.Duration
+	MaxCrawlDuration  time.Duration
+	RenderingTimeout  time.Duration
+	CrawlThreadCount  uint32
+}
+
+func readDashboardPreviewsSettings(iniFile *ini.File) DashboardPreviewsSettings {
+	maxThreadCount := uint32(20)
+
+	s := DashboardPreviewsSettings{}
+
+	previewsCrawlerSection := iniFile.Section("dashboard_previews.crawler")
+	s.CrawlThreadCount = uint32(previewsCrawlerSection.Key("thread_count").MustUint(6))
```

Comment:
I'm guessing the `MustUint(6)` is where we are setting the default values in case there it is undefined / missing in the config file.

More of a general question about how our backend works, but are the `Must` functions part of Golang? (can't seem to find reference to them in Go documentation) I think I understand their purpose here though (ensuring a valid fallback value)

Reply:
The `must` functions are part of the `ini` package: https://pkg.go.dev/gopkg.in/ini.v1